sdk/core/azure_core_test/src/proxy/matchers.rs

```diff
@@ -58,15 +83,35 @@ pub struct CustomDefaultMatcherBody {
     pub ignored_query_parameters: Vec<&'static str>,
 }
 
+impl Default for CustomDefaultMatcher {
+    fn default() -> Self {
+        CustomDefaultMatcher {
+            compare_bodies: None,
+            // TODO: Remove once all recordings are updated (https://github.com/Azure/azure-sdk-for-rust/issues/2411).
+            excluded_headers: vec!["x-ms-client-request-id"],
+            ignored_headers: DEFAULT_IGNORED_HEADERS.to_vec(),
+            ignore_query_ordering: None,
```

Comment:
Shouldn't this be true? We want to sanitize no matter how the querry params appear. Or am I missing how this field works?

Reply:
You're missing how this field works. Not only is this a matcher - not a sanitizer - and query param order should never matter according to HTTP specs, but all this field means is "accept the server default".